pkg/agent/core/ngt/handler/grpc/index_test.go

```diff
@@ -379,88 +388,473 @@ func Test_server_IndexInfo(t *testing.T) {
-		           name: "",
-		           ip: "",
-		           ngt: nil,
-		           eg: nil,
-		           streamConcurrency: 0,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+				if _, err := s.MultiInsert(ctx, req); err != nil {
+					t.Fatal(err)
+				}
+			},
+			want: want{
+				wantRes: &payload.Info_Index_Count{
+					Stored:      0,
+					Uncommitted: insertCnt,
+					Indexing:    false,
+					Saving:      false,
+				},
+			},
+		},
+		{
+			name: "Equivalence Class Testing case 2.3: return uncommitted count with 100 uncommitted delete index",
+			args: args{
+				in1: &payload.Empty{},
+			},
+			fields: fields{
+				name:    name,
+				ip:      ip,
+				svcCfg:  defaultSvcCfg,
+				svcOpts: defaultSvcOpts,
+			},
+			beforeFunc: func(t *testing.T, ctx context.Context, a args, s *server) {
+				// we need to insert request first before remove
+				req, err := request.GenMultiInsertReq(request.Float, vector.Gaussian, insertCnt, dim, defaultInsertConfig)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := s.MultiInsert(ctx, req); err != nil {
+					t.Fatal(err)
+				}
+				if _, err := s.CreateIndex(ctx, &payload.Control_CreateIndexRequest{
+					PoolSize: uint32(len(req.Requests)),
+				}); err != nil {
+					t.Fatal(err)
+				}
+
+				// remove
+				if _, err := s.MultiRemove(ctx, request.GenMultiRemoveReq(removeCnt, defaultRemoveConfig)); err != nil {
+					t.Fatal(err)
+				}
+			},
+			want: want{
+				wantRes: &payload.Info_Index_Count{
+					Stored:      insertCnt,
+					Uncommitted: removeCnt,
+					Indexing:    false,
+					Saving:      false,
+				},
+			},
+		},
+		{
+			name: "Equivalence Class Testing case 2.4: return uncommitted count with 100 uncommitted insert+delete index",
+			args: args{
+				in1: &payload.Empty{},
+			},
+			fields: fields{
+				name:    name,
+				ip:      ip,
+				svcCfg:  defaultSvcCfg,
+				svcOpts: defaultSvcOpts,
+			},
+			beforeFunc: func(t *testing.T, ctx context.Context, a args, s *server) {
+				// we need to insert request first before remove
+				totalInsertCnt := insertCnt + removeCnt
+				req, err := request.GenMultiInsertReq(request.Float, vector.Gaussian, totalInsertCnt, dim, defaultInsertConfig)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := s.MultiInsert(ctx, req); err != nil {
+					t.Fatal(err)
+				}
+				if _, err := s.CreateIndex(ctx, &payload.Control_CreateIndexRequest{
+					PoolSize: uint32(len(req.Requests)),
+				}); err != nil {
+					t.Fatal(err)
+				}
+
+				// remove
+				if _, err := s.MultiRemove(ctx, request.GenMultiRemoveReq(removeCnt, defaultRemoveConfig)); err != nil {
+					t.Fatal(err)
+				}
+				// insert again
+				req, err = request.GenMultiInsertReq(request.Float, vector.Gaussian, insertCnt, dim, defaultInsertConfig)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := s.MultiInsert(ctx, req); err != nil {
+					t.Fatal(err)
+				}
```

Comment:
In this test case, we want the function return the uncommitted count as 100 insert + 100 remove = total 200.

To implement this case, we need 100 uncommitted insert & remove requests stored in the insert & remove queue correspondingly, so we need to insert again here for uncommitted insert index. 


I have updated the comment and refactored the implementation for easier understanding.

Reply:
The first insert here is only for the uncommitted remove count, because we need to insert before removal.
https://github.com/vdaas/vald/blob/48305d8d8bcbbf74f60e176ec1315ffc9b895c60/pkg/agent/core/ngt/handler/grpc/index_test.go#L630-L642

and the second insert is for the uncommitted insert count.
https://github.com/vdaas/vald/blob/48305d8d8bcbbf74f60e176ec1315ffc9b895c60/pkg/agent/core/ngt/handler/grpc/index_test.go#L656-L663